### PR TITLE
Finish event dispatching

### PR DESCRIPTION
**Pull request by arv**
This implementes stopPropagation, stopImmediatePropagation and fixes issues
with AT_TARGET.

### CODE REVIEW

**Comment by arv** [comment]
@sjmiles @sorvell

File: src/wrappers/EventTarget.js [hunk 1]
```diff
@@ -103,72 +105,88 @@
...
-        break;
-      var currentTarget = ancestorChain[i].ancestor;
-      var target = ancestorChain[i].target;
-      if (!deliverEvent(target, currentTarget, event, type, phase))
-        break;
+    if (!propagationStopped) {
+      propagationStopped = invoke(ancestorChain[0], event, type, phase);
+      i++;
+    }
+
```
Comment (dfreedm) [inline src/wrappers/EventTarget.js:121]: Non bubbling events like `focus` need to fire in `AT_TARGET` phase on the ShadowRoot's host, which is `ancestorChain[length-1]` right?

Comment (arv) [inline src/wrappers/EventTarget.js:121]: The target is at `ancestorChain[0]`.

I started implementing retargetting. I'll get to 'focus' eventually.

Comment (dfreedm) [inline src/wrappers/EventTarget.js:121]: SGTM, I'll pull this now then

**Pull request merged**